src/lexer.py

```diff
@@ -328,7 +332,8 @@ def lex(source: str) -> datatypes.TokenStream:
                 raise datatypes.LexerError("Unclosed comment", location=location, start_idx=start_idx, end_idx=idx)
             idx += 1  # skip ')'
 
-        elif current_char.isnumeric() or current_char == '-' or current_char == '.':
+        elif current_char.isnumeric() or current_char == '-' \
+                or (current_char == '.' and source[idx + 1].isnumeric()):
```

Comment:
I would prefer the break come after the or (which should get rid of the line continuation) and with this many conditions, each or should be on its own line.